### PR TITLE
Added admin panel

### PR DESCRIPTION
**Pull request by RedStylzZ**

### CODE REVIEW

**Review by crazySilvia — APPROVED** [review]
LGTM

File: backend/src/main/java/com/github/redstylzz/backend/service/UserService.java [hunk 1]
```diff
@@ -1,56 +1,55 @@
...
-                    .username(user.getUsername())
-                    .password(new Argon2PasswordEncoder().encode(user.getPassword()))
-                    .rights(List.of("USER"))
-                    .accountNonLocked(true)
-                    .accountNonExpired(true)
-                    .credentialsNonExpired(true)
-                    .enabled(true)
-                    .build();
+    public String addUser(MongoUser admin, MongoUserDTO user) throws ResponseStatusException {
+        LOG.debug(user);
```
Comment (Flooooooooooorian) [inline backend/src/main/java/com/github/redstylzz/backend/service/UserService.java:31]: Dont log the password

File: backend/src/main/java/com/github/redstylzz/backend/service/UserService.java [hunk 2]
```diff
@@ -1,56 +1,55 @@
...
+                        .accountNonExpired(true)
+                        .credentialsNonExpired(true)
+                        .enabled(true)
+                        .build();
 
-            repository.save(newUser);
-            LOG.debug("Added new User: " + newUser);
-            return jwtService.createToken(newUser);
+                repository.save(newUser);
+                LOG.debug("Added new User: " + newUser);
```
Comment (Flooooooooooorian) [inline backend/src/main/java/com/github/redstylzz/backend/service/UserService.java:46]: same

File: frontend/src/components/RequireAdmin.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import {ReactElement, useContext} from "react";
+import {DataContext} from "../context/DataProvider";
+
+export default function RequireAdmin({children}: { children: ReactElement<any, any> }) {
+    const {isAdmin} = useContext(DataContext)
+
+    console.log("IsAdmin:", isAdmin)
```
Comment (Flooooooooooorian) [inline frontend/src/components/RequireAdmin.tsx:7]: remove console.log

File: frontend/src/context/DataProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+export interface IDataContextProps {
+    isAdmin?: boolean
+}
+
+export const DataContext = createContext<IDataContextProps>({})
+
+export default function DataProvider({children}: { children: ReactElement<any, any> }) {
+    const {config, isLoggedIn} = useContext(AuthContext)
+    const controller: IUserController = useMemo(() => UserController(config), [config])
+    const [isAdmin, setIsAdmin] = useState<boolean | undefined>(undefined)
```
Comment (Flooooooooooorian) [inline frontend/src/context/DataProvider.tsx:15]: Why undefined as type?

File: frontend/src/pages/AdminPage.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    const [password, setPassword] = useState<string>("")
+    const [worked, setWorked] = useState<string>("")
+    const [role, setRole] = useState<string>("USER")
+    const controller: IUserController = UserController(config)
+
+    const addUser: FormEventHandler<HTMLFormElement> = (event) => {
+        event.preventDefault()
+
+        if ((username && username.length) && (password && password.length) && (role && role.length)) {
+            console.log("Add User:", username, password, role)
```
Comment (Flooooooooooorian) [inline frontend/src/pages/AdminPage.tsx:19]: remove console.log

File: frontend/src/pages/AdminPage.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    const onUsernameChange = (event: ChangeEvent<HTMLInputElement>) =>
+        setUsername(event.target.value)
+
+    const onPasswordChange = (event: ChangeEvent<HTMLInputElement>) =>
+        setPassword(event.target.value)
+
+    const onRoleChange = (event: ChangeEvent<HTMLInputElement>) => {
+        console.log(event.target.value)
```
Comment (Flooooooooooorian) [inline frontend/src/pages/AdminPage.tsx:39]: remove

File: frontend/src/pages/AdminPage.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            const user: User = {
+                username: username,
+                password: password,
+                rights: [role]
+            }
+            controller.addUser(user).then(response => {
+                const status: string = response ? "Success" : "Failed"
+                setWorked(status)
+            })
+        }
```
Comment (Flooooooooooorian) [inline frontend/src/pages/AdminPage.tsx:29]: else {} Show the user an error

File: frontend/src/services/UserService.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+export default function UserService(config: ITokenConfig | undefined): IUserController {
+
+    return {
+        addUser: (user) => {
+            return axios.post("/api/user/", user, config).then(response => response.data)
+        },
+        isAdmin: () => {
+            return axios.get("/api/user/admin", config).then(response => {
+                console.log(response.data)
```
Comment (Flooooooooooorian) [inline frontend/src/services/UserService.ts:13]: remove

File: frontend/src/context/DataProvider.tsx (hunk 4 above)
Comment (RedStylzZ) [reply frontend/src/context/DataProvider.tsx:15]: Is required for a third status (Not Logged in)

**Pull request merged**